src/fam.rs

```diff
@@ -169,9 +169,11 @@ impl<T: Default + FamStruct> FamStructWrapper<T> {
     ///
     /// Get the capacity required by mem_allocator in order to hold
     /// the provided number of [`FamStruct::Entry`](trait.FamStruct.html#associatedtype.Entry).
-    fn mem_allocator_len(fam_len: usize) -> usize {
-        let wrapper_size_in_bytes = size_of::<T>() + fam_len * size_of::<T::Entry>();
-        (wrapper_size_in_bytes + size_of::<T>() - 1) / size_of::<T>()
+    /// Returns `None` if the required length would overflow usize.
+    fn mem_allocator_len(fam_len: usize) -> Option<usize> {
+        let wrapper_size_in_bytes =
+            size_of::<T>().checked_add(fam_len.checked_mul(size_of::<T::Entry>())?)?;
+        Some((wrapper_size_in_bytes.checked_add(size_of::<T>() - 1)?) / size_of::<T>())
```

Comment:
This will panic if `size_of::<T>() == 0`, so it should be:
```suggestion
        Some((wrapper_size_in_bytes.checked_add(size_of::<T>().checked_sub(1)?)?).checked_div(size_of::<T>())?)
```
There may be a way to re-work this to be a bit nicer.

Reply:
True! I guess if you try to use this with zero-sized types then all hope is lost anyway, but yeah, lets do the proper checks here. I left it at "?-forest" for now, will think about a nicer way to do this after lunch :)